cli/cmd/debug/bpf/features_linux.go

```diff
@@ -0,0 +1,61 @@
+package bpf
+
+import (
+	"fmt"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/features"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+// featuresCmd outputs available BPF features on the host
+var featuresCmd = &cobra.Command{
+	Use:   "features",
+	Short: "Output available BPF features on the host",
+	RunE: func(*cobra.Command, []string) error {
+		linuxVersion, err := features.LinuxVersionCode()
+		if err != nil {
+			return errors.Wrap(err, "failed to get Linux version code")
+		}
+		fmt.Printf("Linux kernel version: %s\n", getLinuxKernelVersion(linuxVersion))
+
+		fmt.Println("--------------------------------------------------")
+
+		err = features.HaveBoundedLoops()
+		if err != nil && !errors.Is(err, ebpf.ErrNotSupported) {
+			return errors.Wrap(err, "failed to check for bounded loops")
+		}
+		fmt.Printf("Bounded loops: %s\n", isSupported(err))
+
+		fmt.Println("--------------------------------------------------")
+
+		err = features.HaveLargeInstructions()
+		if err != nil && !errors.Is(err, ebpf.ErrNotSupported) {
+			return errors.Wrap(err, "failed to check for large instructions")
+		}
+		fmt.Printf("Large instructions: %s\n", isSupported(err))
+
+		fmt.Println("--------------------------------------------------")
+		fmt.Println("eBPF map availability:")
+		for _, mt := range eBPFMapList {
+			err = features.HaveMapType(mt)
+			if err != nil && !errors.Is(err, ebpf.ErrNotSupported) {
+				return errors.Wrapf(err, "failed to check for map type %s", mt.String())
+			}
+			fmt.Printf("%s: %s\n", mt.String(), isSupported(err))
+		}
+
+		fmt.Println("--------------------------------------------------")
+		fmt.Println("eBPF program types availability:")
+		for _, pt := range eBPFProgramList {
+			err = features.HaveProgramType(pt)
+			if err != nil && !errors.Is(err, ebpf.ErrNotSupported) {
+				return errors.Wrapf(err, "failed to check for program type %s", pt.String())
+			}
+			fmt.Printf("%s: %s\n", pt.String(), isSupported(err))
+		}
```

Comment:
IMO, it's not important to list all map/prog types. We can just list all maps and programs, and that would be enough information. For specific maps (like conntrack), we may want to print entries, but listing by types is not that helpful.

Reply:
this is more so for users to check which maps and programs can be initialize on their host, but yes i do plan to have a command that will list out maps/programs running